Sources/Examples/ExploringPerformanceExample.swift

```diff
@@ -0,0 +1,201 @@
+import Foundation
+import Logging
+
+// MARK: Example library code
+
+struct DataContainer {
+    let data: String // this would be some user `T` type
+
+    // would be W3C's `traceparent` for example
+    let traceContext: TraceContext?
+}
+
+
+// there's a few ways to specify those, this is just an example;
+//
+// See others: https://github.com/opentracing/specification/blob/master/rfc/trace_identifiers.md
+// Though this one is going to be the standardized one soon: https://w3c.github.io/trace-context/
+struct TraceContext {
+    let version: UInt8
+    let traceId: (UInt64, UInt64) // one value in 128 bits
+    let parentId: UInt64
+    let flags: UInt8
+}
+
+extension TraceContext: CustomStringConvertible {
+    public var description: String {
+        let v = String(self.version, radix: 16, uppercase: true)
+        let t1 = String(self.traceId.0, radix: 16, uppercase: true)
+        let t2 = String(self.traceId.1, radix: 16, uppercase: true)
+        let s = String(self.parentId, radix: 16, uppercase: true)
+        let f = String(self.flags, radix: 16, uppercase: true)
+
+        // rendering just an example, did not super closely follow any spec, but close enough:
+        return "\(v)-\(t1)\(t2)-\(s)-\(f)"
+    }
+}
+
+struct SomeContext {
+    let id: String
+    var log: Logger
+}
+
+class Library {
+
+    // somewhere in library, per context
+    // imagine we initialize it safely somewhere, always once,
+    // it has the right context metadata (e.g. the above id), which remains the same for the lifetime of this context
+    // but also we need to set things on it per invocation
+    private var log: Logger
+
+    // pseudo structure
+    private var context: SomeContext
+
+    init() {
+        let id = "some-id"
+        // assume we know the context id here:
+        var log = Logging.make(id)
+        log.metadata = ["id": "\(id)"]
+        log.metadata["additional"] = "some-per-context-info" // could require "rendering into a string"
+
+        self.log = log
+
+        let context: SomeContext = SomeContext(id: id, log: self.log)
+        self.context = context
+    }
+
+    private func extractMeta(from envelope: DataContainer, withBase metadata: LoggingMetadata) -> LoggingMetadata {
+        var renderedMetadata: LoggingMetadata = [:]
+
+        switch envelope.traceContext {
+        case .some(let trace):
+            renderedMetadata = [
+                "per-data0": "some-metadata",
+                "trace": trace // by allowing Any as value, we delay the rendering operation
```

Comment:
this rides on the fact that LoggingMetadata is now `String:Any` so we don't have to invoke the rendering of the trace yet (nor any other values which would have to run some more code to render than a plain "expose my fields").

With a `String:String` metadata container we'd be forced to:

```swift
        var renderedMetadata: LoggingMetadata = [:]

         switch envelope.traceContext {
         case .some(let trace):
             renderedMetadata = [
                 "per-data0": "some-metadata",
                 // TODO: So that's the thing I'm a bit worried about with the string metadata:
                 // TODO: this rendering is a bit costly, though we'd likely want to carry around the integers rather than the string repr...
                 // TODO: Sadly with String metadata we always have to render it, even if we don't even log here AT ALL :-(
                 // TODO: solution 1) pass around the trace metadata as String all the time in the envelope, this
                 // TODO: solution 2) "whatever", yeah it costs
                 "trace": "\(trace)" // runs . description eagerly, even if we'd never log
             ]
```